crates/runtime/src/gc/enabled.rs

```diff
@@ -206,17 +207,15 @@ impl Drop for VMExternRef {
     fn drop(&mut self) {
```

Comment:
Oh and now I also realize that these same concerns apply to `Clone` above, that seems like a footgun that it's safe since there's no easy way to audit callers are all happening in the same thread.

Reply:
Yeah, it is definitely a footgun, and this isn't how I would design this from scratch. It is just the intermediate stepping stone on the way to the next PR, which introduces a proper `GcHeap` that is passed in to GC hooks/barriers which are proof of safe access to the heap and its objects (similar to what store does at the embedder API level).

I don't really know how to avoid this intermediate state without fusing this PR together with the next.

Reply:
Ok if this is just a stepping stone I think that's ok. If you're ok with it though I think it'd be best at this point to land this PR tomorrow at the earliest since the 19.0.0 branch point will happen tonight and I think it would be best to not have this pair of PRs get straddled across a release

Reply:
That makes sense to me.